packages/core/src/oidc/init.ts

```diff
@@ -377,6 +377,39 @@ export default function initOidc(
       throw error;
     }
   });
+  /**
+   * Check if the request URL contains comma separated `resource` query parameter. If yes, split the values and
+   * reconstruct the URL with multiple `resource` query parameters.
+   * E.g. `?resource=resource1,resource2` => `?resource=resource1&resource=resource2`
+   */
+  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
+  oidc.use(async (ctx, next) => {
+    const { query } = ctx.request;
+    const { resource } = query;
+
+    let processedResource = resource;
+
+    if (Array.isArray(resource)) {
+      processedResource = resource.reduce((accumulator: string[], value: string) => {
+        if (value.includes(',')) {
+          return [...accumulator, ...value.split(',')];
+        }
+
+        return [...accumulator, value];
+      }, []);
+    }
+
+    if (typeof resource === 'string' && resource.includes(',')) {
+      processedResource = resource.split(',');
+    }
```

Comment:
hit:

```ts
const resources = Array.isArray(resource) ? resource : [resource];

const resourceParam = resource.reduce((accumulator: string[], value: string) => {
    if (value.includes(',')) {
      return [...accumulator, ...value.split(',')];
    }

    return [...accumulator, value];
  }, []);
```

Reply:
Updated. Extracted to a middleware file, and also added test cases